test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperRunnerTest.cs

```diff
@@ -88,5 +161,25 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 output.Attributes["foo"] = context.AllAttributes["foo"].ToString();
             }
         }
+
+        public class OrderedTagHelper : TagHelper
+        {
+            public OrderedTagHelper(int order)
+            {
+                Order = order;
+            }
+
+            public override int Order { get; }
+            public IList<OrderedTagHelper> ProcessOrderTracker { get; set; }
```

Comment:
simplify to `IList<int>` or an `Action<int>`.  nothing needs the tag helpers themselves so either is an improvement.  delegate leaves list management to the test, but that's probably minor.